core/server/src/main/java/alluxio/StreamCache.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+import com.google.common.cache.RemovalNotification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Cache for storing file input and output streams.
+ *
+ * When a stream is added to the cache, the cache returns a unique ID that can be used for future
+ * lookups of the stream. The streams are automatically closed when their cache entry is
+ * invalidated or after being inactive for an extended period of time.
+ */
+@ThreadSafe
+public final class StreamCache {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static final RemovalListener<Integer, Closeable> CLOSER =
+      new RemovalListener<Integer, Closeable>() {
+        public void onRemoval(RemovalNotification<Integer, Closeable> removal) {
+          try {
+            removal.getValue().close();
+          } catch (Exception e) {
+            LOG.error("Failed to close stream: ", e);
+          }
+        }
+      };
+
+  private AtomicInteger mCounter;
+  private Cache<Integer, FileInStream> mInStreamCache;
+  private Cache<Integer, FileOutStream> mOutStreamCache;
+
+  /**
+   * Creates a new instance of {@link StreamCache}.
+   */
+  public StreamCache() {
```

Comment:
can we take the timeout as a parameter and have the caller look it up in the configuration?